### PR TITLE
Csl632 recv timeout

### PR DESCRIPTION
**Pull request by georgeee**

### CODE REVIEW

File: constants.production.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ntpPollDelay: 300000000 # 300 sec
+
+# Update mechanism
+updateServers: []
+updateProposalThreshold: 0.1 # 10% of total stake
+updateVoteThreshold: 0.001 # 0.1% of total stake
+updateImplicitApproval: 40000 # slots
+
+# Production start time
+productionNetworkStartTime: 0
```
Comment (georgeee) [inline constants.production.yaml]: This file was removed from master. Probably merge conflict resolved inappropriately. Need to remove it here as well

File: src/Pos/Util.hs [hunk 2]
```diff
@@ -704,16 +706,40 @@ withWaitLogConvL nodeId conv = conv { send = send', recv = recv' }
...
+    :: ( Mockable Async m
+       , Mockable Delay m
+       , Mockable Bracket m
+       )
+    => Microsecond -> m a -> m (Maybe a)
 execWithTimeLimit timeout action = do
     promises <- mapM async [ Just <$> action, delay timeout $> Nothing ]
-    (promise, val) <- waitAny promises
-    mapM_ cancel $ filter (/= promise) promises
+    (_, val) <- waitAny promises `finally` mapM_ cancel promises
```
Comment (georgeee) [inline src/Pos/Util.hs]: Is `cancel` an idempotential operation?

File: src/Pos/Communication/Server.hs [hunk 3]
```diff
@@ -64,6 +76,18 @@ allStubListeners p =
         , usStubListeners
         ]
 
+  where
```
Comment (georgeee) [inline src/Pos/Communication/Server.hs]: Code within `where` is not used and should be removed

**Comment by georgeee** [comment]
Also, tests failed with timeout on Travis. Could you check they run properly locally?

File: src/Pos/Util.hs (hunk 2 above)
Comment (Martoon-00) [inline src/Pos/Util.hs]: Probably noo, but it does nothing for completed operations

File: src/Pos/Communication/Server.hs (hunk 3 above)
Comment (Martoon-00) [inline src/Pos/Communication/Server.hs]: Mda :facepalm:

File: src/Pos/Util.hs (hunk 2 above)
Comment (georgeee) [inline src/Pos/Util.hs]: Yeah, that what I wanted to know :)

**Pull request merged**